src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int AMOUNT_OF_FIGURES = 5;
+    private static final int MAX_RANDOM_VALUE = 100;
+    private Random random = new Random();
+    private int numOfFigure;
+    private String color;
+    private ColorSupplier colorSupplier;

```

Comment:
Services we rely on are better not to initialize with new in some method as they may be very heavy and we will spend lot of resources creating them each time we call method
At the same time numOfFigure and color are used in 1 method only so we may easily use them as method variables

```suggestion
    private final Random random = new Random();
    private final ColorSupplier colorSupplier = new ColorSupplier();
```